website/files/models/osfstorage.py

```diff
@@ -108,6 +108,9 @@ def is_checked_out(self):
         return self.checkout is not None
 
     def delete(self, user=None, parent=None):
+        if self.node.preprint_file == self:
+            self.node._is_preprint_orphan = True
```

Comment:
If we wanted the providers to stay in tact after the "Restore preprint" pipeline is implemented - than no I don't think so! But we could.